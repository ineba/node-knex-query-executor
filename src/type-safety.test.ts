import spawn from 'cross-spawn'
import path from 'path'

test('Typescript', () => {
    const typescriptCompilation = spawn.sync('./node_modules/.bin/tsc', [
        '-p',
        path.resolve(__dirname, './type-safety-fixtures/testconfig.json'),
    ])

    const output = typescriptCompilation.stdout.toString()

    expect(output).toMatchSnapshot('Typescript expected failures')
})
