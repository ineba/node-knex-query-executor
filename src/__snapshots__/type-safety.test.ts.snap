// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typescript: Typescript expected failures 1`] = `
"src/type-safety-fixtures/args-respected.ts(37,27): error TS7006: Parameter '_' implicitly has an 'any' type.
src/type-safety-fixtures/args-respected.ts(45,41): error TS2345: Argument of type '1' is not assignable to parameter of type 'object'.
src/type-safety-fixtures/args-respected.ts(46,41): error TS2345: Argument of type '1' is not assignable to parameter of type 'string'.
src/type-safety-fixtures/args-respected.ts(47,41): error TS2345: Argument of type '{}' is not assignable to parameter of type '{ id: number; }'.
  Property 'id' is missing in type '{}' but required in type '{ id: number; }'.
src/type-safety-fixtures/args-respected.ts(48,41): error TS2345: Argument of type '{}' is not assignable to parameter of type '{ id: number; }'.
  Property 'id' is missing in type '{}' but required in type '{ id: number; }'.
src/type-safety-fixtures/args-respected.ts(50,11): error TS2554: Expected 2 arguments, but got 1.
src/type-safety-fixtures/create-query-helper.ts(17,20): error TS2339: Property 'wrongTable' does not exist on type 'TableNames<\\"testTable\\">'.
src/type-safety-fixtures/create-query-helper.ts(18,16): error TS2339: Property 'wrongTable' does not exist on type 'Tables<\\"testTable\\">'.
src/type-safety-fixtures/create-query-helper.ts(20,21): error TS2339: Property 'foo' does not exist on type '{ testArg: string; }'.
src/type-safety-fixtures/query-arguments.ts(22,37): error TS2345: Argument of type '{}' is not assignable to parameter of type '{ testArg: string; }'.
  Property 'testArg' is missing in type '{}' but required in type '{ testArg: string; }'.
"
`;
